docs/src/man/comparisons.md

```diff
@@ -53,61 +59,116 @@ df[findall(==('c'), df.id), :]
-| Rename column after aggregation | `df.groupby('grp')['x'].mean().reset_index().rename(columns={'x': 'mean_x'})`             | `combine(groupby(df, :grp), :x => mean => :mean_x)` |
-| Aggregate and add columns       | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')`                       | `transform(groupby(df, :grp), :x => mean)`          |
-| Aggregate and select columns    | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')[['grp','x_mean','y']]` | `select(groupby(df, :grp), :x => mean, :y)`         |
+DataFrames.jl provides a `groupby` function such that one can perform analytic functions
+over each group independently. The result of `groupby` is a `GroupedDataFrame` object
+which may be processed using the `combine`, `transform`, or `select` functions.
+The following table illustrate some common grouping and aggregation usages.
+
+| Operation                       | pandas                                                                                | DataFrames.jl                                        |
+|---------------------------------|---------------------------------------------------------------------------------------|------------------------------------------------------|
+| Aggregate by groups             | `df.groupby('grp')['x'].mean()`                                                       | `combine(groupby(df, :grp), :x => mean)`             |
+| Rename column after aggregation | `df.groupby('grp')['x'].mean().rename("my_mean")`                                     | `combine(groupby(df, :grp), :x => mean => :my_mean)` |
+| Add aggregated data as column   | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')`                   | `transform(groupby(df, :grp), :x => mean)`           |
+| ...and select output columns    | `df.join(df.groupby('grp')['x'].mean(), on='grp', rsuffix='_mean')[['grp','x_mean']]` | `select(groupby(df, :grp), :id, :x => mean)`         |
+
+Note that pandas returns a `Series` object for 1-dimensional result, so the corresponding
+DataFrames.jl examples are made to return an equivalent `DataFrame` object. Consider the
+first example:
+
+```python
+>>> df.groupby('grp')['x'].mean()
+grp
+1    4
+2    3
+Name: x, dtype: int64
+```
+
+For DataFrames.jl, it looks like this:
+
+```julia
+julia> combine(groupby(df, :grp), :x => mean)
+2×2 DataFrame
+│ Row │ grp   │ x_mean  │
+│     │ Int64 │ Float64 │
+├─────┼───────┼─────────┤
+│ 1   │ 1     │ 4.0     │
+│ 2   │ 2     │ 3.0     │
+```
+
+Pandas supports hierarchical indexing. DataFrames.jl provides similar functionality with
+the `groupby` function. For example, if you want to drill down the data frame with `grp` equals
+`1` and then `x` equals `4`, then we can write the following code:
+
+```julia
+@pipe df |> groupby(_, :grp)[(1,)] |> groupby(_, :x)[(4,)]
```

Comment:
This isn't the most efficient nor logical way to write this: better use `filter` instead. The text should explain that `groupby` is useful for efficient *repeated* indexing.

Also, I'd rather avoid introducing `@pipe` here. Let's leave that to the DataFramesMeta page to avoid making things too complex.

Reply:
Are you talking about using `filter` on the original data frame?  In that case, I would just remove this section altogether and simplify this section. See commit https://github.com/JuliaData/DataFrames.jl/pull/2378/commits/51addbc06666528db71ba889a717d1b785948360